examples/get-started/react/google-maps/app.jsx

```diff
@@ -13,86 +12,71 @@ const AIR_PORTS =
 const GOOGLE_MAP_ID = process.env.GoogleMapsMapId; // eslint-disable-line
 
-const renderMap = status => {
-  if (status === Status.LOADING) return <h3>{status} ..</h3>;
-  if (status === Status.FAILURE) return <h3>{status} ...</h3>;
-  return null;
-};
-
-function MyMapComponent({center, zoom}) {
-  const ref = useRef();
-  const [map, setMap] = useState(null);
-  const overlay = useMemo(
-    () =>
-      new DeckOverlay({
-        layers: [
-          new GeoJsonLayer({
-            id: 'airports',
-            data: AIR_PORTS,
-            // Styles
-            filled: true,
-            pointRadiusMinPixels: 2,
-            pointRadiusScale: 2000,
-            getPointRadius: f => 11 - f.properties.scalerank,
-            getFillColor: [200, 0, 80, 180],
-            // Interactive props
-            pickable: true,
-            autoHighlight: true,
-            onClick: info =>
-              info.object &&
-              // eslint-disable-next-line
-              alert(`${info.object.properties.name} (${info.object.properties.abbrev})`)
-          }),
-          new ArcLayer({
-            id: 'arcs',
-            data: AIR_PORTS,
-            dataTransform: d => d.features.filter(f => f.properties.scalerank < 4),
-            // Styles
-            getSourcePosition: f => [-0.4531566, 51.4709959], // London
-            getTargetPosition: f => f.geometry.coordinates,
-            getSourceColor: [0, 128, 200],
-            getTargetColor: [200, 0, 80],
-            getWidth: 1
-          })
-        ]
-      }),
-    []
-  );
+function DeckGLOverlay(props) {
+  const map = useMap();
+  const overlay = useMemo(() => new DeckOverlay(props));
 
   useEffect(() => {
     if (map) {
-      map.setCenter(center);
-      map.setZoom(zoom);
       overlay.setMap(map);
+      return () => overlay.setMap(null);
     }
-  }, [map, center, zoom, overlay]);
+    return undefined;
+  }, [map])
 
-  useEffect(() => {
-    const mapInstance = new window.google.maps.Map(ref.current, {
-      mapId: GOOGLE_MAP_ID
-    });
-    setMap(mapInstance);
-  }, []);
-  return (
-    <>
-      <div ref={ref} id="map" style={{height: '100vh', width: '100wh'}} />
-    </>
-  );
+  overlay.setProps(props);
+  return null;
 }
 
 function Root() {
-  const center = {lat: 51.47, lng: 0.45};
-  const zoom = 5;
+  const onClick = info => {
+    if (info.object) {
+      // eslint-disable-next-line
+      alert(`${info.object.properties.name} (${info.object.properties.abbrev})`);
+    }
+  };
+
+  const layers = [
+    new GeoJsonLayer({
+      id: 'airports',
+      data: AIR_PORTS,
+      // Styles
+      filled: true,
+      pointRadiusMinPixels: 2,
+      pointRadiusScale: 2000,
+      getPointRadius: f => 11 - f.properties.scalerank,
+      getFillColor: [200, 0, 80, 180],
+      // Interactive props
+      pickable: true,
+      autoHighlight: true,
+      onClick,
+      // beforeId: 'waterway-label' // In interleaved mode render the layer under map labels
```

Comment:
Does the google overlay support `beforeId`? If it does, the layer name for Google is probably different than for Mapbox.

Reply:
no it doesn't